dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/dto/resources/visitor/ResourceTreeVisitor.java

```diff
@@ -123,9 +124,15 @@ private static ResourceComponent getResourceComponent(StorageEntity resource) {
             tempResourceComponent = new FileLeaf();
         }
 
+        String currentDir = "";
+        if (resource.getFullName().contains(Constants.RESOURCE_TYPE_FILE)) {
+            currentDir = resource.getFullName().split(Constants.RESOURCE_TYPE_FILE)[1].substring(1);
+        }
```

Comment:
This is not safe, we should parse the `ResourceMetaData` and check whether the `resource'resourceType` is `FILE`. You can find some code under `AbstractStorageOperator#getResourceMetaData`

Reply:
Done.